pkg/runtime/processor/processor_test.go

```diff
@@ -654,3 +654,51 @@ func TestReporter(t *testing.T) {
 			}
 		})
 }
+
+func TestProcessorWaitGroupError(t *testing.T) {
+	ctx := context.Background()
+	proc, _ := newTestProc()
+	// spin up the processor
+	go func() {
+		proc.Process(ctx)
+	}()
+
+	comp1 := componentsapi.Component{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testpubsub1",
+		},
+		Spec: componentsapi.ComponentSpec{
+			Type:         "pubsub.mockPubSub",
+			Version:      "v1",
+			Metadata:     daprt.GetFakeMetadataItems(),
+			InitTimeout:  "2",
+			IgnoreErrors: true,
+		},
+	}
+	comp2 := componentsapi.Component{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testpubsub2",
+		},
+		Spec: componentsapi.ComponentSpec{
+			Type:         "pubsub.mockPubSub",
+			Version:      "v1",
+			Metadata:     daprt.GetFakeMetadataItems(),
+			InitTimeout:  "2",
+			IgnoreErrors: true,
+		},
+	}
+	time.Sleep(1 * time.Second)
```

Comment:
```suggestion
```

Reply:
sleep is kinda lame but is the easy way of ensuring the processor is receiving on the channel

Reply:
What difference does that make?